### PR TITLE
Add PR template

### PR DESCRIPTION
**Pull request by lkostrowski**
I'd like to introduce 1st PR template that could help writing better description and checklist for required steps

### CODE REVIEW

**Comment by peelar** [comment]
What do you think about reusing [the templates in `saleor/saleor`](https://github.com/saleor/saleor/tree/main/.github/ISSUE_TEMPLATE)?

**Comment by lkostrowski** [comment]
@peelar I think its PR template https://github.com/saleor/saleor/blob/main/.github/PULL_REQUEST_TEMPLATE.md

But it's specific for Saleor/Saleor backend logic

**Comment by peelar** [comment]
Ah sorry, wrong link. What I meant is that I think we should try to introduce a common structure to the PR/issue templates in our main repositories.

**Comment by lkostrowski** [comment]
@peelar I agree in general but I don't think we can achieve it at the moment. There are not only very different types (lib vs app vs cli, client vs internal, frontend vs backend), but also very different maturity level. Points from other repositories might not really be valid here. We also don't have established our final DoD so it's incremental

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+**I want to merge this PR, because: ***
```
Comment (peelar) [inline .github/PULL_REQUEST_TEMPLATE.md:1]:
```suggestion
**I want to merge this PR because: ***
```

**Review by peelar — APPROVED** [review]

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]
From my experience, markdown templates are mostly filled improperly (not everyone knows how to use markdown checkboxes for example).

I would suggest using YAML form templates https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

Experience of using them is much better and a chance to use the wrong syntax is minimized

**Comment by lkostrowski** [comment]
@krzysztofwolski this is great but it's only supported for issues :/

**Review by krzysztofwolski — APPROVED** [review]
Hope YAML templates will land on PRs in the future :c

**Comment by lkostrowski** [comment]
@krzysztofwolski afaik it's on the roadmap, we will introduce it as soon as its shipped ;)

**Comment by michalina-graczyk** [comment]
Not need to be in QA.

**Pull request merged**